jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/snmp/SnmpSession.java

```diff
@@ -252,74 +252,17 @@ public void endSession() throws IOException {
-      // --> 1.4.1.7.3 = "010114111530"
-      // rsuSRMDeliveryStop.3 x "010114130000"
-      // --> 1.4.1.8.3 = "010114130000"
-      // rsuSRMPayload.3 x "0EFF82445566778899000000AABBCCDDEEFF00E00EA0C12A00"
-      // --> 1.4.1.9.3 = "0EFF82445566778899000000AABBCCDDEEFF00E00EA0C12A00"
-      // rsuSRMEnable.3 = 1
-      // --> 1.4.1.10.3 = 1
-      //////////////////////////////
-
-      VariableBinding rsuSRMPsid = SnmpSession.getPEncodedVariableBinding(
-            "1.0.15628.4.1.4.1.2.".concat(Integer.toString(index)),
-            snmp.getRsuid());
-      VariableBinding rsuSRMDsrcMsgId = new VariableBinding(
-            new OID("1.0.15628.4.1.4.1.3.".concat(Integer.toString(index))), new Integer32(snmp.getMsgid()));
-      VariableBinding rsuSRMTxMode = new VariableBinding(
-            new OID("1.0.15628.4.1.4.1.4.".concat(Integer.toString(index))), new Integer32(snmp.getMode()));
-      VariableBinding rsuSRMTxChannel = new VariableBinding(
-            new OID("1.0.15628.4.1.4.1.5.".concat(Integer.toString(index))), new Integer32(snmp.getChannel()));
-      VariableBinding rsuSRMTxInterval = new VariableBinding(
-            new OID("1.0.15628.4.1.4.1.6.".concat(Integer.toString(index))), new Integer32(snmp.getInterval()));
-      VariableBinding rsuSRMDeliveryStart = new VariableBinding(
-            new OID("1.0.15628.4.1.4.1.7.".concat(Integer.toString(index))),
-            new OctetString(DatatypeConverter.parseHexBinary(SNMP.snmpTimestampFromIso(snmp.getDeliverystart()))));
-      VariableBinding rsuSRMDeliveryStop = new VariableBinding(
-            new OID("1.0.15628.4.1.4.1.8.".concat(Integer.toString(index))),
-            new OctetString(DatatypeConverter.parseHexBinary(SNMP.snmpTimestampFromIso(snmp.getDeliverystop()))));
-      VariableBinding rsuSRMPayload = new VariableBinding(
-            new OID("1.0.15628.4.1.4.1.9.".concat(Integer.toString(index))),
-            new OctetString(DatatypeConverter.parseHexBinary(payload)));
-      VariableBinding rsuSRMEnable = new VariableBinding(
-            new OID("1.0.15628.4.1.4.1.10.".concat(Integer.toString(index))), new Integer32(snmp.getEnable()));
-      VariableBinding rsuSRMStatus = new VariableBinding(
-            new OID("1.0.15628.4.1.4.1.11.".concat(Integer.toString(index))), new Integer32(snmp.getStatus()));
-
-      ScopedPDU pdu = new ScopedPDU();
-      pdu.add(rsuSRMPsid);
-      pdu.add(rsuSRMDsrcMsgId);
-      pdu.add(rsuSRMTxMode);
-      pdu.add(rsuSRMTxChannel);
-      pdu.add(rsuSRMTxInterval);
-      pdu.add(rsuSRMDeliveryStart);
-      pdu.add(rsuSRMDeliveryStop);
-      pdu.add(rsuSRMPayload);
-      pdu.add(rsuSRMEnable);
-      if (verb == ServiceRequest.OdeInternal.RequestVerb.POST) {
-         pdu.add(rsuSRMStatus);
+   public static ScopedPDU createPDU(SNMP snmp, String payload, int index, RequestVerb verb, SnmpProtocol snmpProtocol) throws ParseException {
+      if (snmpProtocol == SnmpProtocol.FOURDOT1) {
```

Comment:
Switch statements in java are faster than if-statements in most cases and they are significantly faster if there are a lot of cases. We don't have many cases here but it couldn't hurt to use a switch statement. It would be much more readable, too. I've replaced the if-statement here with a switch statement.